packages/frontend/i18n/src/utils/time.ts

```diff
@@ -0,0 +1,281 @@
+  year: 7,
+} satisfies Record<TimeUnit, number>;
+
+/**
+ * ```ts
+ * // timestamp to string
+ * i18nTime(1728538228000) -> 'Oct 10, 2024, 1:30:28 PM'
+ *
+ * // absolute time string
+ * i18nTime('2024-10-10 13:30:28', { absolute: { accuracy: 'minute' } }) -> '2024-10-10 13:30 PM'
+ * i18nTime('2024-10-10 13:30:28', { absolute: { accuracy: 'minute', noDate: true } }) -> '13:30 PM'
+ * i18nTime('2024-10-10 13:30:28', { absolute: { accuracy: 'minute', noYear: true } }) -> 'Oct 10, 13:30 PM'
+ * i18nTime('2024-10-10 13:30:28', { absolute: { accuracy: 'day' } }) -> 'Oct 10, 2024'
+ *
+ * // relative time string
+ * i18nTime('2024-10-10 13:30:30', { relative: true }) -> 'now'
+ * i18nTime('2024-10-10 13:30:00', { relative: true }) -> '30s ago'
+ * i18nTime('2024-10-10 13:30:30', { relative: { accuracy: 'minute' } }) -> '2m ago'
+ *
+ * // show absolute time string if time is pass 1 day
+ * i18nTime('2024-10-9 14:30:30', { relative: { max: [1, 'day'] } }) -> '23h ago'
+ * i18nTime('2024-10-9 13:30:30', { relative: { max: [1, 'day'] } }) -> 'Oct 9, 2024, 1:30:30 PM'
+ * ```
+ */
+export function i18nTime(
+  time: dayjs.ConfigType,
+  options: {
+    // override i18n instance, default is global I18n instance
+    i18n?: I18n;
+    // override now time, default is current time
+    now?: dayjs.ConfigType;
+    relative?:
+      | {
+          // max time to show relative time, if time is pass this time, show absolute time
+          max?: [number, TimeUnit];
+          // show time with this accuracy
+          accuracy?: TimeUnit;
+          // show weekday, e.g. 'Monday', 'Tuesday', etc.
+          weekday?: boolean;
+          // show 'yesterday' or 'tomorrow' if time is
+          yesterdayAndTomorrow?: boolean;
+        }
+      | true; // use default relative option
+    absolute?: {
+      // show time with this accuracy
+      accuracy?: TimeUnit;
+      // hide year
+      noYear?: boolean;
+      // hide date (year, month, day)
+      noDate?: boolean;
+    };
+  } = {}
+) {
+  const i18n = options.i18n ?? I18n;
+  time = dayjs(time);
+  const now = dayjs(options.now);
+
+  const defaultRelativeOption = {
+    max: [1000, 'year'],
+    accuracy: 'second',
+    weekday: false,
+    yesterdayAndTomorrow: true,
+  } satisfies typeof options.relative;
+
+  const relativeOption = options.relative
+    ? options.relative === true
+      ? defaultRelativeOption
+      : {
+          ...defaultRelativeOption,
+          ...options.relative,
+        }
+    : null;
+
+  const defaultAbsoluteOption = {
+    accuracy: 'second',
+    noYear: false,
+    noDate: false,
+  } satisfies typeof options.absolute;
+
+  const absoluteOption = {
+    ...defaultAbsoluteOption,
+    ...options.absolute,
+  };
+
+  if (relativeOption) {
+    // show relative
+
+    const formatter = new Intl.RelativeTimeFormat(i18n.language, {
+      style: 'narrow',
+      numeric: relativeOption.yesterdayAndTomorrow ? 'auto' : 'always',
+    });
+
+    const timeUnitProcessor = {
```

Comment:
I think dayjs already provides relative time with locale support https://day.js.org/docs/en/plugin/relative-time
Do we need to do it on our own?

Reply:
dayjs needs to load additional language files to implement i18n, using browser intl api can reduce the bundle size